### ISSUE TITLE
Unexpected @graphql-eslint/known-fragment-names when using fragments within fragments

### ISSUE THREAD

**Issue by beaucollins** [dotansimha/graphql-eslint#654]
**Describe the bug**

Not precisely sure what the exact requirements are but this is close to what we have in our codebase. It appears that a certain depth of fragment use in an `operations` document is not able to be found. I tried moving the fragment definitions in different orders with the same result.

Moving the fragments into the same file as the operation that uses them will cause the error to go away.

Also the line numbers that are reported are the wrong file and not the correct location if they were in the other file.

https://github.com/beaucollins/graphql-lint-fragment-variables/runs/3715892638

**To Reproduce**

```
git clone -b fragment-not-found git@github.com:beaucollins/graphql-lint-fragment-variables.git
npm install
npm run lint
```

```
graphql-lint-fragment-variables/src/admin/allOwners.graphql
  11:9  error  Unknown fragment "TimelineEntry"  @graphql-eslint/known-fragment-names
  14:9  error  Unknown fragment "TimelineEntry"  @graphql-eslint/known-fragment-names
```

**Expected behavior**

The fragment should be found and validated correctly. Copying the contents of `sharedFields.graphql` into `allOwners.graphql` shows the expected result.

**Environment:**

- OS: macOS 11.5.2
- `@graphql-eslint/eslint-plugin@2.2.0`
- NodeJS: v16.9.1

**Additional context**

**Comment by B2o5T** [comment]
I think there is a problem with `@graphql-tools/import` package, but not with `graphql-eslint`. I'll investigate this

**Comment by B2o5T** [comment]
fixed in `@graphql-tools@6.6.0`, will be fixed also in the next minor version of `graphql-eslint`

**Comment by dotansimha** [comment]
Fixed in @graphql-eslint/eslint-plugin@2.5.0

**Comment by beaucollins** [comment]
I'm still seeing this problem using `@graphql-eslint/eslint-plugin@2.5.0` and `@graphq-tools/import@6.6.1`.

**Comment by beaucollins** [comment]
Here's the build with `2.5.0` and `6.6.1` with the same eslint error:

https://github.com/beaucollins/graphql-lint-fragment-variables/runs/4219647556?check_suite_focus=true

**Comment by B2o5T** [comment]
I tested it with default import, but forget to test with named import, I think we need adjust a fix for named export too

**Comment by FreekVR** [comment]
I am actually running into the same issues in my project, using a default and not a named import, with eslint-plugin@latest :(

**Comment by B2o5T** [comment]
@FreekVR I need your reproduction repo because the current reproduction repo show 0 errors on the latest version of `graphql-eslint`
Also we removed the need of `#import` comments in favor of `parserOptions.operations` in [3.3.0 version](https://github.com/dotansimha/graphql-eslint/releases/tag/%40graphql-eslint%2Feslint-plugin%403.3.0)

**Comment by B2o5T** [comment]
Closing as the original issue was fixed on `graphql-eslint`. @FreekVR I'll reopen an issue if you still need help
